src/tamp_improv/approaches/improvisational/policies/rl.py

```diff
@@ -161,10 +167,17 @@ def train(self, env: gym.Env, train_data: TrainingData) -> None:
         # Train the model
         self.model.learn(total_timesteps=total_timesteps, callback=callback)
 
-        print("\nFinal Training Results:")
-        print(f"Overall Success Rate: {callback.final_success_rate:.2%}")
-        print(f"Overall Avg Episode Length: {callback.final_avg_episode_length:.2f}")
-        print(f"Overall Avg Reward: {callback.final_avg_reward:.2f}")
+        if (
```

Comment:
this `hasattr` business is a little hacky -- can we instead just put the printing inside the callback itself?